crates/core/src/vm.rs

```diff
@@ -447,10 +447,17 @@ impl<'db, 'tx> DbProgram<'db, 'tx> {
                 // TODO: How do we deal with mutating values?
                 Table::MemTable(_) => Err(ErrorVm::Other(anyhow::anyhow!("How deal with mutating values?"))),
                 Table::DbTable(x) => {
+                    let inserts = rows.clone();
```

Comment:
Would prefer not to clone here, but db.insert wants a row, not a &row. Is it worth extending the refactor in there too?